src/app/models/form.server.model.ts

```diff
@@ -208,7 +211,69 @@ const compileFormModel = (db: Mongoose): IFormModel => {
             'Check that your form is MyInfo-authenticated, is an email mode form and has 30 or fewer MyInfo fields.',
         },
       },
-      form_logics: [LogicSchema],
+      form_logics: {
+        type: [LogicSchema],
+        validate: {
+          validator(v: ILogicSchema[]) {
+            /**
+             * `this` variable points to the parent document object when we use
+             * save().
+             */
+            const form = this as unknown as IFormSchema
+
+            /**
+             * A condition object is said to be validatable if it contains the two
+             * necessary for validation: fieldType and state
+             */
+            type ValidatableCondition = IncompleteValidatableCondition & {
+              fieldType: BasicField
+            }
+
+            /**
+             * A validatable condition is incomplete if there is a possibility
+             * that its fieldType is null, which is a sign that a condition's
+             * field property references a non-existent form_field.
+             */
+            type IncompleteValidatableCondition = {
+              state: LogicConditionState
+              fieldType?: BasicField
+            }
+
+            const isAllConditionsReferenceExistingFields = (
+              conditions: IncompleteValidatableCondition[],
+            ): conditions is ValidatableCondition[] =>
+              !conditions.some((condition) => !condition.fieldType)
+
+            const conditions = v.flatMap((logic) => {
+              return logic.conditions.map<IncompleteValidatableCondition>(
+                (condition) => {
+                  const {
+                    field,
+                    state,
+                  }: { field: ObjectId; state: LogicConditionState } = condition
```

Comment:
ooo I'm a bit wary of asserting that `field` is an `ObjectId`. it's not inconceivable that some of them in the database have been saved as strings, so calling `toHexString` on them would result in a null pointer exception. can we use the `String` constructor instead so we can deal with both ObjectIds and strings?